### ISSUE TITLE
Fulltextsearch is indexing too much

### ISSUE THREAD

**Issue by AljosaB** [tractorcow-farm/silverstripe-fluent#497]
Hi,

I have two locales - en_US (default) and de_DE where there is plenty of en_US pages and 0 de_DE pages (translations). There are **no fallbacks** set for any locale. If I run fulltextsearch reindex I end up having all the en_US pages in Solr which is great. The problem is there is also the same number of de_DE pages with english content and __locale field set to de_DE. That would be OK if de_DE had a fallback to en_US but it doesn't. In this case there sould be no de_DE records in Solr at all. I believe FluentSearchVariant needs to be fixed in a way to exclude pages without translations if there is no fallback.

**Comment by robbieaverill** [comment]
Yes, this sounds like a bug. Thanks for raising the issue.

**Comment by AljosaB** [comment]
I looked into it and found a workaround / dirty fix

**FluentSeachVariant.php**

```
public function activateState($state)
{
	$fluentState = FluentState::singleton()->setLocale($state);
	$fluentState->setIsFrontend(true); // workaround
}
```

**Comment by robbieaverill** [comment]
To be fair, I'm not sure that's a dirty fix at all. Solr content is usually used on the frontend, so it's likely that this is actually the correct approach to fixing the issue. Would you like to make a pull request?

**Comment by tractorcow** [comment]
Ah yes that's right. The SolrContext should be setting frontend to true for reindexing. It should probably be fixed in the fulltextsearch module itself, rather than in here.

**Comment by robbieaverill** [comment]
Fixed in #511
